pylint/test/test_self.py

```diff
@@ -113,6 +114,7 @@ def _runtest(self, args, reporter=None, out=None, code=None):
 
     def _run_pylint(self, args, out, reporter=None):
         args = args + ["--persistent=no"]
+        sys.argv = ["pylint"] + args
```

Comment:
Why do we add `pylint` here?

Reply:
Because sys.argv[0] is used in the Usage help text, which is checked by one of my tests.
Instead of setting sys.argv here, we could set it only for the test(s) that need it or we simply only check the return value of the pylint call (via the code kwargs of _runtests). I would prefer the latter.

Reply:
Yeah, the latter sounds better for me as well.